docs/design/dd-003-step-by-step.md

```diff
@@ -922,29 +938,29 @@ Whatever choice we make, it should probably include some form of
 dependency injection for a trace that allows us to collect the events we
 care about more precisely and with less effort.
 
-### Step-by-Step Measurements
+### 2.3. Stepwise Measurements
```

Comment:
I think we should continue to call this `step-by-step` given that I have used this name everywhere in many commits and the document is also called `step-by-step`.

Reply:
I will revert this change. My rationale was that, on a first glimpse, it can be a bit confusing that one of the studied tactics and the rewrite proposal is called the same (so I renamed the references in the section to describe the tactic), but it does also make sense to keep it consistent.

Reply:
You're raising a very good point. For sure, I didn't initially parse your intention when reading your diff. So, I spent some extra time considering your point. In the end, I still think this is fine because (a) we never actually implemented a step-by-step technique and (b) the proposal seems to me is just proposing to apply such a technique using context-based-injection as the implementation mechanism to separate netxlite and measurexlite.